### PR TITLE
FIX: untested attributesignal attributeerror exception

### PR DESCRIPTION
**Pull request by klauer**
Don't merge yet - still am not getting coverage

### CODE REVIEW

File: tests/test_device.py [hunk 1]
```diff
@@ -235,6 +236,17 @@ def prop(self, value):
...
 
 
+def test_attribute_signal_attributeerror():
+    class MyDevice(Device):
+        sig1 = Component(AttributeSignal, 'unknownattribute')
+        sig2 = Component(AttributeSignal, '__class__.__name')
+
+    dev = MyDevice('', name='mydev')
+    pytest.raises(AttributeError, dev.sig1.get)
+    pytest.raises(AttributeError, getattr(dev.sig1, 'base'))
```
Comment (tacaswell) [inline tests/test_device.py]: I think this needs to be `pytest.raises(AtributeError, getattr, dev.sig1, 'base')`?

Comment (klauer) [inline tests/test_device.py]: I do this all the time - it's annoying

**Comment by klauer** [comment]
This took me more (distracted) tries than I'd like to admit. Ready for review.

File: ophyd/signal.py [hunk 2]
```diff
@@ -918,8 +918,7 @@ def base(self):
             try:
                 obj = getattr(obj, part)
             except AttributeError as ex:
-                attr = '.'.join(self.parent_attr[:i + 1])
-                raise AttributeError('{} ({})'.format(attr, ex))
+                raise AttributeError('{} ({})'.format(part, ex))
```
Comment (tacaswell) [inline ophyd/signal.py]: :bike: :house: is it worth including `obj` here as well?

**Pull request merged**